### PR TITLE
[Modules] Updated AAD Domain Services module to new dependency approach

### PR DESCRIPTION
**Pull request by AlexanderSehr**
# Description

- Updated AAD Domain Services module to new dependency approach

| Pipeline |
| - |
| [AAD: DomainServices](https://github.com/Azure/ResourceModules/actions/workflows/ms.aad.domainservices.yml)  <= As expected |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: modules/Microsoft.AAD/DomainServices/.test/common/deploy.test.bicep [hunk 1]
```diff
@@ -0,0 +1,87 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for a testing purposes.')
```
Comment (MrMCake) [inline modules/Microsoft.AAD/DomainServices/.test/common/deploy.test.bicep:6]:
```suggestion
@description('Optional. The name of the resource group to deploy for testing purposes.')
```

**Comment by AlexanderSehr** [comment]
Finally. The deployment works up to the point we expect (i.e. all dependencies work and are in the correct format)

File: modules/Microsoft.AAD/DomainServices/.test/common/dependencies.bicep [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+}
+
+@description('The resource ID of the created Virtual Network Subnet.')
+output subnetResourceId string = virtualNetwork.properties.subnets[0].id
+
+@description('The resource ID of the created Key Vault.')
+output keyVaultResourceId string = keyVault.id
+
+@description('The name of the certification password secret.')
+output CertPWSecretName string = CertPWSecretName
```
Comment (eriqua) [inline modules/Microsoft.AAD/DomainServices/.test/common/dependencies.bicep:97]:
```suggestion
output certPWSecretName string = certPWSecretName
```

File: modules/Microsoft.AAD/DomainServices/.test/common/dependencies.bicep [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+@description('The name of the certification secret.')
+output CertSecretName string = CertSecretName
```
Comment (eriqua) [inline modules/Microsoft.AAD/DomainServices/.test/common/dependencies.bicep:100]:
```suggestion
output certSecretName string = certSecretName
```

File: modules/Microsoft.AAD/DomainServices/.test/common/dependencies.bicep [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+@description('Required. The name of the Key Vault to create.')
+param keyVaultName string
+
+@description('Required. The name of the Managed Identity to create.')
+param managedIdentityName string
+
+@description('Required. The name of the Deployment Script to create for the Certificate generation.')
+param certDeploymentScriptName string
+
+var CertPWSecretName = 'pfxCertificatePassword'
```
Comment (eriqua) [inline modules/Microsoft.AAD/DomainServices/.test/common/dependencies.bicep:16]:
```suggestion
var certPWSecretName = 'pfxCertificatePassword'
```

File: modules/Microsoft.AAD/DomainServices/.test/common/dependencies.bicep [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+var CertSecretName = 'pfxBase64Certificate'
```
Comment (eriqua) [inline modules/Microsoft.AAD/DomainServices/.test/common/dependencies.bicep:17]:
```suggestion
var certSecretName = 'pfxBase64Certificate'
```

File: modules/Microsoft.AAD/DomainServices/.test/common/deploy.test.bicep [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    domainName: '<<namePrefix>>.onmicrosoft.com'
+    additionalRecipients: [
+      '<<namePrefix>>@noreply.github.com'
+    ]
+    diagnosticLogsRetentionInDays: 7
+    diagnosticStorageAccountId: diagnosticDependencies.outputs.storageAccountResourceId
+    diagnosticWorkspaceId: diagnosticDependencies.outputs.logAnalyticsWorkspaceResourceId
+    diagnosticEventHubAuthorizationRuleId: diagnosticDependencies.outputs.eventHubAuthorizationRuleId
+    diagnosticEventHubName: diagnosticDependencies.outputs.eventHubNamespaceEventHubName
+
```
Comment (eriqua) [inline modules/Microsoft.AAD/DomainServices/.test/common/deploy.test.bicep:75]:
```suggestion
```

File: modules/Microsoft.AAD/DomainServices/.test/common/deploy.test.bicep [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+    lock: 'CanNotDelete'
+    pfxCertificate: keyVault.getSecret(resourceGroupResources.outputs.CertSecretName)
```
Comment (eriqua) [inline modules/Microsoft.AAD/DomainServices/.test/common/deploy.test.bicep:77]:
```suggestion
    pfxCertificate: keyVault.getSecret(resourceGroupResources.outputs.certSecretName)
```

File: modules/Microsoft.AAD/DomainServices/.test/common/deploy.test.bicep [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+    pfxCertificatePassword: keyVault.getSecret(resourceGroupResources.outputs.CertPWSecretName)
```
Comment (eriqua) [inline modules/Microsoft.AAD/DomainServices/.test/common/deploy.test.bicep:78]:
```suggestion
    pfxCertificatePassword: keyVault.getSecret(resourceGroupResources.outputs.certPWSecretName)
```

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**